tools/wptrunner/wptrunner/testrunner.py

```diff
@@ -31,6 +30,25 @@ def release_mozlog_lock():
                                  'browser_cls', 'browser_kwargs'])
 
 
+class StopState(NamedTuple):
+    """Synchronization for coordinating a graceful exit.
+
+    Attributes:
+        should_stop: Flag that is polled by threads so that they can gracefully
+            exit in the face of SIGINT.
+        done: A barrier that each `TestRunnerManager` thread waits on when exiting
+            its run loop. This provides a reliable way for the `ManagerGroup` to
+            tell when all threads have cleaned up their resources.
+    """
+    should_stop: threading.Event
+    done: threading.Barrier
```

Comment:
I am not good at pick names, but maybe the easier choice is to call it `_barrier`, as it is an object of Barrier.